scipy/cluster/hierarchy.py

```diff
@@ -165,7 +165,7 @@ def _copy_array_if_base_present(a):
     """
     if a.base is not None:
         return a.copy()
-    elif np.issubsctype(a, np.float32):
+    elif issubclass(a.dtype.type, np.float32):
```

Comment:
Is this now a canonic NumPy way of checking if an array is single-precision? In the same like, is the way to check if an array is floating-point (as opposed to integer or complex), `issubclass(a.dtype.type, np.floating)`?

Asking because there is also `np.issubdtype` --- would be great to actually know what's the preferred way now that numpy 2.0 is on the horizon

Reply:
Yeah, not sure, but I think folks smarter than I debate this type of stuff in the issue I linked?

Reply:
Ah indeed. Reading comprehension issue on my side, sorry about that!

Agreed, let the bright minds of numpy converge to something, we'll happily follow. The current spelling is certainly better than `issubsctype`, so +1 from me.
(I probably could have a tiny preference for `np.issubdtype` but it's not worth the churn).